### PR TITLE
Added dosctrings for ivy.astype

### PR DESCRIPTION
**Pull request by rensortino**
When I first tested the function to generate the docstring examples, it raised the following exceptions:
1. `TypeError: astype() got an unexpected keyword argument 'out'`
    which I solved by adding the out argument in the corresponding numpy backend function (if needed, before merging the PR I can add it also to the other backends).
2. `TypeError: Cannot cast array data from dtype('float64') to dtype('int32') according to the rule 'same_kind'`, which was a problem of the `np.copyto` function called inside the numpy `inplace_update`, which I solved by casting the array before giving it to input to `np.copyto`.

### CODE REVIEW

**Comment by rensortino** [comment]
Solves #2020

**Comment by RitujaPawas** [comment]
@rensortino please add docstrings and docstring examples for `astype()` in `ivy/array/data_type` as well as in `ivy/container/data_type`

**Comment by RitujaPawas** [comment]
Make sure to request a review when you have made the changes!

**Comment by RitujaPawas** [comment]
There might be an issue with requesting a review. Thanks for letting me know. I will review ASAP!

**Comment by RitujaPawas** [comment]
@rensortino  please fix the [lint-issues](https://github.com/unifyai/ivy/runs/7653345152?check_suite_focus=true)  and [test-docstrings issues](https://github.com/unifyai/ivy/runs/7653345018?check_suite_focus=true) related to data_type.py and specific to your function!

File: ivy/array/data_type.py [hunk 1]
```diff
@@ -10,6 +10,53 @@ class ArrayWithDataTypes(abc.ABC):
...
+            shape as ``x``.
+
+        Examples
+        --------
+
+        Using :code:`ivy.Array` instance method:
+
+        >>> x = ivy.array([[-1, -2], [0, 2]])
+        >>> x.astype(ivy.float64)
+        ivy.array([[-1., -2.],  [0.,  2.]])
```
Comment (RitujaPawas) [inline ivy/array/data_type.py:58]: Add a print statement for this.

File: ivy/container/data_type.py [hunk 2]
```diff
@@ -22,6 +22,54 @@ def static_astype(
...
+        ret
+            an array having the specified data type. The returned array must have the same
+            shape as ``x``.
+
+        Examples
+        --------
+
+        >>> c = ivy.Container(a=ivy.array([False,True,True]), b=ivy.array([3.14, 2.718, 1.618]))
+        >>> ivy.Container.static_astype(c, int)
+        {
```
Comment (RitujaPawas) [inline ivy/container/data_type.py:67]: For dtype, please use ivy.{dtype}

File: ivy/container/data_type.py [hunk 3]
```diff
@@ -45,6 +93,57 @@ def astype(
...
+        --------
+
+        Using :code:`ivy.Container` instance method:
+        
+        >>> x = ivy.Container(a=ivy.array([False,True,True]), \
+                                b=ivy.array([3.14, 2.718, 1.618]))
+        >>> x.astype(ivy.int32)
+        {
+            a: ivy.array([0, 1, 1]),
+            b: ivy.array([1, 0, 1])
```
Comment (RitujaPawas) [inline ivy/container/data_type.py:143]: make sure the output is correct, and add print statement for printing outputs

**Review by RitujaPawas — COMMENTED** [review]
make sure you add print statements for printing outputs! and please re-check the values of output.

**Comment by rensortino** [comment]
Hi, after the last commits I checked the tests and there shouldn't be any more related to the `astype` function, as far as it concerns the docstring and lint tests. I hope everything's right, please let me know if there is anything else to fix.

**Comment by RitujaPawas** [comment]
Sure @rensortino  sorry for the delay. I will take a look and let you know ASAP

**Review by RitujaPawas — CHANGES_REQUESTED** [review]
Hi @rensortino  can you please pull the latest changes from master and request review again ASAP. Everything looks good.

**Review by RitujaPawas — APPROVED** [review]
hey @rensortino this looks good! Thank you for your contributions!

**Pull request merged**